src/pybind/mgr/dashboard/frontend/src/app/ceph/block/iscsi-target-form/iscsi-target-form.component.ts

```diff
@@ -0,0 +1,463 @@
+import { Component } from '@angular/core';
+import { FormArray, FormControl, Validators } from '@angular/forms';
+import { Router } from '@angular/router';
+
+import { I18n } from '@ngx-translate/i18n-polyfill';
+import * as _ from 'lodash';
+import { BsModalRef, BsModalService } from 'ngx-bootstrap/modal';
+import { forkJoin } from 'rxjs';
+
+import { IscsiService } from '../../../shared/api/iscsi.service';
+import { RbdService } from '../../../shared/api/rbd.service';
+import { SelectMessages } from '../../../shared/components/select/select-messages.model';
+import { SelectOption } from '../../../shared/components/select/select-option.model';
+import { CdFormGroup } from '../../../shared/forms/cd-form-group';
+import { CdValidators } from '../../../shared/forms/cd-validators';
+import { FinishedTask } from '../../../shared/models/finished-task';
+import { TaskWrapperService } from '../../../shared/services/task-wrapper.service';
+import { IscsiTargetImageSettingsModalComponent } from '../iscsi-target-image-settings-modal/iscsi-target-image-settings-modal.component';
+import { IscsiTargetIqnSettingsModalComponent } from '../iscsi-target-iqn-settings-modal/iscsi-target-iqn-settings-modal.component';
+
+@Component({
+  selector: 'cd-iscsi-target-form',
+  templateUrl: './iscsi-target-form.component.html',
+  styleUrls: ['./iscsi-target-form.component.scss']
+})
+export class IscsiTargetFormComponent {
+  targetForm: CdFormGroup;
+  modalRef: BsModalRef;
+  minimum_gateways = 1;
+  target_default_controls: any;
+  disk_default_controls: any;
+
+  imagesAll: any[];
+  imagesSelections: SelectOption[];
+  portalsSelections: SelectOption[] = [];
+  imagesInitiatorSelections: SelectOption[] = [];
+  groupDiskSelections: SelectOption[] = [];
+  groupMembersSelections: SelectOption[] = [];
+  imagesSettings: any = {};
+  messages = {
+    portals: new SelectMessages(
+      { noOptions: this.i18n('There are no portals available.') },
+      this.i18n
+    ),
+    images: new SelectMessages(
+      { noOptions: this.i18n('There are no images available.') },
+      this.i18n
+    ),
+    initiatorImage: new SelectMessages(
+      {
+        noOptions: this.i18n(
+          'There are no images available. Please make sure you add an image to the target.'
+        )
+      },
+      this.i18n
+    ),
+    groupInitiator: new SelectMessages(
+      {
+        noOptions: this.i18n(
+          'There are no initiators available. Please make sure you add an initiator to the target.'
+        )
+      },
+      this.i18n
+    )
+  };
+
+  IQN_REGEX = /^iqn\.(19|20)\d\d-(0[1-9]|1[0-2])\.\D{2,3}(\.[A-Za-z0-9-]+)+(:[A-Za-z0-9-\.]+)*$/;
```

Comment:
[A-Za-z] will limit the characters to ASCII only. What about Unicode? Is this limited to ASCII by design? If Unicode is not supported, please add a comment.

Reply:
I copied this regex from openATTIC, so I expect it to be correct.